python/ql/src/semmle/python/web/django/Request.qll

```diff
@@ -67,7 +67,7 @@ private class DjangoView extends ClassValue {
 }
 
 private FunctionValue djangoViewHttpMethod() {
-    exists(DjangoView view | view.attr(httpVerbLower()) = result)
+    exists(DjangoView view | view.declaredAttribute(httpVerbLower()) = result)
```

Comment:
:thinking: Why `declaredAttribute` rather than `attr`? The former is much more restrictive in where it looks for the given attribute.

Reply:
This came from the fact that using `attr` gave almost no results. I remember we talked about this, and concluded you would look into this. I don't think we made any tickets about this.

For the actual code, we could use `lookup` as well, which is not as restricted as `declaredAttribute`. However, since we only use this to identify the `request` parameter on functions that will handle http request in view-classes, I do feel like we only care about functions declared in classes that inherit from `django.views.generic.View`

However, now I'm wondering which if that correct. If we have this code, what will happen on a POST request? :thinking: 

If there is a respones from `Foo`, we should use `lookup`, but if not, I stand by my decision to use `declaredAttribute`.

```
from django.views.generic import View
from django.http.response import HttpResponse

class Foo(object):
    def post(self, request):
        return HttpResponse('post from Foo')

class ClassView(View, Foo):
    def get(self, request):
        return HttpResponse('get from ClassView')
```

I'm going to try this out!

Reply:
welp, I got a response from Foo when I send a POST request